pgmpy/utils/compat_fns.py

```diff
@@ -165,3 +165,55 @@ def allclose(arr1, arr2, atol):
             arr2,
             atol=atol,
         )
+
+
+def zeros(size, dtype=None):
+    if config.get_backend() == "numpy":
+        return np.zeros(size, dtype=dtype or config.get_dtype())
+    else:
+        return torch.zeros(
+            size, dtype=dtype or config.get_dtype(), device=config.get_device()
+        )
+
+
+def random_choice(values, size=1, p=None, seed=None):
+    if seed is not None:
```

Comment:
Please use `np.random.default_rng` and `torch.random.fork_rng` to set seeds only for this method. Directly calling `np.random.seed` or `manual_seed` would set a global seed for them.